src/identity/manager.rs

```diff
@@ -355,36 +355,28 @@ impl<T: CertificateProvider> SecretManager<T> {
                 drop(certs);
                 // Notify the background worker to start refreshing the certificate.
                 match self.requests.send((id.clone(), pri)).await {
-                    // Note that this should not happen, the background worker is expected to be
-                    // running.
-                    //
-                    // TODO: Better error propagation.
-                    Err(_) => Err(Error::EmptyResponse(id.clone())),
+                    Err(e) => unreachable!("SecretManager worker died: {e}"),
```

Comment:
can you explain why self.requests.send cannot be recovered?

Reply:
That means the worker has died, so this and *all* further calls will fail with this error. Basically the SecretManager instance is useless. You can try to recover from it by re-creating the SecretManager, but it's just silencing a programming error in a `try..catch(anything)` fashion. Rust has good support for resource ownership and makes it hard to maintain global state so it's less of a risk, but I have seen this kind of approach cause deadlocks in other languages.

Reply:
IC, this is requests send channel.